rpg.py

```diff
@@ -41,56 +54,136 @@ def deal_damage(self, other_player):
+                targeted_index = all_enemy_locations[0]
+                targeted_player = players[targeted_index]
+                damaged_player = self.deal_damage(targeted_player)
+                players[targeted_index] = damaged_player
         return players
 
 
 class Ally(Character):
     # subclass of character
     """This is a subclass of characters specific to team 1"""
-    def __init__(self, name: str, hp: int, attack: int, speed: int):
-        super().__init__(name, hp, attack, speed, team=1, level=1, exp=0, target_exp=200)
+    def __init__(self,
+                 name: str,
+                 hp: int,
+                 max_hp: int,
+                 attack: int,
+                 speed: int):
+        super().__init__(name,
+                         hp,
+                         max_hp,
+                         attack,
+                         speed,
+                         team=1,
+                         level=1,
+                         exp=0,
+                         target_exp=200)
 
     def act(self, players):
         """the act method specific to team 1"""
-        # I want to input the name of the enemy to attack
+
         all_enemy_locations = self.get_all_enemies(players)
         enemies = {character.name:
                    character for character in players if character.team != 1}
         if all_enemy_locations:
             if self.is_alive:
-                print(f'{self.name}\'s turn!')
-                print('Who do you attack?')
-                for player in players:
-                    if not player.is_ally(self) and player.is_alive():
-                        print(player.name)
-                targeted_enemy = None
-                while targeted_enemy is None:
-                    user_input = input('>')
-                    targeted_enemy = enemies.get(user_input)
-                damaged_player = self.deal_damage(targeted_enemy)
+                if all_enemy_locations.count(1):
+                    targeted_index = all_enemy_locations[0]
+                    targeted_player = players[targeted_index]
+                    damaged_player = self.deal_damage(targeted_player)
+                    players[targeted_index] = damaged_player
+                else:
+                    targeted_enemy = None
+                    while targeted_enemy is None:
+                        user_input = input('>')
+                        targeted_enemy = enemies.get(user_input)
+                    damaged_player = self.deal_damage(targeted_enemy)
 
         return players
 
+    def move_input(self):
+        move = None
+        while move is None:
+            m_input = input('>')
+            move = m_input
+            if move == 'f':
+                pass
+            elif move == 'i':
+                self.use_item()
+            elif move == 's':
+                self.get_stats()
+                self.move_input()
+            else:
+                self.move_input()
+            return move
```

Comment:
This is really hacky and defeats the point on a while loop. You are actually using recursion.

You don't need a while loop at all, or you need to commit to the while loop and return `move` outside of the loop.